notebooks/NIRISS_WFSS_advanced/01_niriss_wfss_image2_image3.ipynb

```diff
@@ -0,0 +1,1025 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "7a935d7c",
+   "metadata": {},
+   "source": [
+    "# Run Image Pipeline and Create Source Catalog\n",
+    "\n",
+    "In this run, we are starting from the rate files which have already been calibrated with the detector1 pipeline. This will save time as we do not need to edit any of the steps being performed as part of detector1. Therefore, the first calibration that should be done as part of a WFSS run is to run the rate files direct images through the Image2 and Image3 steps of the JWST pipeline. This includes creating a source catalog, which most likely will need to be adjusted from the pipeline default values. **Not having a good source catalog will result in non optimal extraction of sources in the dispersed, WFSS, images.**\n",
+    "\n",
+    "**Use case**: The default parameters for the pipeline do not extract the expected sources, so a custom parameters need to be set to obtain new combined image and source catalog.<br>\n",
+    "**Data**: JWST/NIRISS images and spectra from program 2079.<br>\n",
+    "**Tools**: astropy, glob, jdaviz, json, jwst, matplotlib, numpy, os, pandas, warnings<br>\n",
+    "**Cross-instrument**: NIRISS<br>\n",
+    "\n",
+    "**Content**\n",
+    "- [Imports & Data Setup](#imports)\n",
+    "- [Default Imaging Pipeline Run](#default)\n",
+    "  - [Image2](#default_image2)\n",
+    "  - [Image3](#default_image3)\n",
+    "  - [Inspecting Default Results](#view_default)\n",
+    "- [Custom Imaging Pipeline Run](#custom)\n",
+    "  - [Image3](#custom_image3)\n",
+    "  - [Inspecting Custom Results](#view_custom)\n",
+    "- [Refining the Source Catalog Further](#source_cat)\n",
+    "  - [Matching Source IDs Across Catalogs](#match_sources) \n",
+    "  - [Manually Editing the Source Catalog](#manual_cat)\n",
+    "\n",
+    "**Author**: Rachel Plesha (rplesha@stsci.edu), Camilla Pacifici (cpacifici@stsci.edu), JWebbinar notebooks.<br>\n",
+    "**First Published**: January 2024"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7fdaf1a1-57e5-4239-b82e-6552261d8d82",
+   "metadata": {},
+   "source": [
+    "<a id='imports'></a>\n",
+    "## Imports & Data Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "028352f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import glob\n",
+    "import json\n",
+    "import warnings\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import astropy.units as u\n",
+    "from astropy.io import fits\n",
+    "from astropy.coordinates import SkyCoord\n",
+    "from astropy.table import Table\n",
+    "\n",
+    "from jdaviz import Imviz\n",
+    "from matplotlib import pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import jwst\n",
```

Comment:
Setting the crds path should happen before importing jwst or it will be ignored.